tests/iterator.rs

```diff
@@ -75,3 +91,122 @@ fn signals_block_wait() {
         .expect_err("Wait didn't wait properly");
     assert_eq!(err, RecvTimeoutError::Timeout);
 }
+
+#[test]
+#[serial]
+fn pending_doesnt_block_and_is_empty_if_no_signals_recieved() {
+    let signals = setup_for_sigusr2();
+
+    let mut recieved_signals = signals.pending();
+
+    assert_eq!(recieved_signals.next(), None);
+}
+
+#[test]
+#[serial]
+fn wait_returns_recieved_signals() {
+    let signals = setup_for_sigusr2();
+    send_sigusr2();
+
+    let mut recieved_signals = signals.wait();
+
+    assert_eq!(recieved_signals.next(), Some(SIGUSR2));
+    assert_eq!(recieved_signals.next(), None);
```

Comment:
I think it would be more convenient/readable to use something like `assert_eq(vec![SIGUSR2], received_signals.collect())` here and at other places. It is a bit less performant (allocating to the vec), but that shouldn't be a problem for tests.